tasks/install.yaml

```
@@ -0,0 +1,34 @@
+- block:
+
+  - name: Add MariaDB repository key
+    apt_key:
+      url: "{{ mariadb.repository.apt.key_url }}"
+
+  - name: Add MariaDB repository
+    apt_repository:
+      repo: "{{ mariadb.repository.apt.repository }}"
+      filename: /etc/apt/sources.list.d/mariadb.list
+
+  when: ansible_distribution == 'Ubuntu'
+
+- name: update apt cache
+  apt:
+    update_cache: yes
+  when: ansible:distribution == 'Ubuntu'
```

* Fix typo in `ansible:distribution`
* Only update apt cache if necessary
* Move task to block as it uses the same `when` condition
* Fix task name

```suggestion
- when: ansible_distribution == 'Ubuntu'
  block:
    - name: Add MariaDB repository key
      apt_key:
        url: "{{ mariadb.repository.apt.key_url }}"

    - name: Add MariaDB repository
      apt_repository:
        repo: "{{ mariadb.repository.apt.repository }}"
        filename: /etc/apt/sources.list.d/mariadb.list
      register: mariadb_add_apt_repository

    - name: Update apt cache
      apt:
        update_cache: yes
      when: mariadb_add_apt_repository.changed
```